src/Filesystem/Provider/FileAccess.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\DTO\FileSize;
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+
+/**
+ * Interface FileAccess
+ *
+ * The FileAccess interface defines all file operations.
+ * Filesystem role interface.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ *
+ * @public
+ */
+interface FileAccess {
```

Comment:
@chfsx If I see something depending on `FileRead` I know it won't write. I would expect more things that only read. `FileWrite` seems to be more dangerous in general.